### PR TITLE
message protection kat

### PR DESCRIPTION
**Pull request by franziskuskiefer**
This adds the message protection kat.
It's not entirely finished bute generally passing. There are still a few todos in the kat that I added to #1279.

Note that this also makes the change from mlswg/mls-protocol#866.

### CODE REVIEW

File: openmls/src/group/core_group/mod.rs [hunk 1]
```diff
@@ -909,7 +909,7 @@ impl CoreGroup {
         };
 
         // Build AuthenticatedContent
-        let mut commit = AuthenticatedContent::commit(
+        let mut authenticated_content = AuthenticatedContent::commit(
```
Comment (raphaelrobert) [inline openmls/src/group/core_group/mod.rs:912]: Just double-checking: is this renaming intended or some leftover from earlier?

**Review by raphaelrobert — APPROVED** [review]
Just one question. Feel free to merge otherwise.

File: openmls/src/group/core_group/mod.rs (hunk 1 above)
Comment (franziskuskiefer) [reply openmls/src/group/core_group/mod.rs:869]: It's part of me debugging issues. And I found the name here confusing.
I'll click the auto-merge button. Please request change if you want it changed back.

**Comment by franziskuskiefer** [comment]
Looks like #1280 broke some APIs. So this will have to wait.

**Pull request merged**